channels/printer/client/win/printer_win.c

```diff
@@ -432,6 +435,19 @@ rdpPrinterDriver* win_freerdp_printer_client_subsystem_entry(void)
 		win_driver->driver.ReleaseRef = printer_win_release_ref_driver;
 
 		win_driver->id_sequence = 1;
+		win_driver->defaultPrinter = NULL;
+
+		GetDefaultPrinter(NULL, &size);
+		if (size)
+		{
+			win_driver->defaultPrinter = (LPWSTR)calloc(size, sizeof(WCHAR));
+
+			if (win_driver->defaultPrinter)
```

Comment:
If we can't allocate memory for the default printer's name, I'd rather exit with an error and log a message, than continuing like if everything is ok..

Reply:
So what about:

		win_driver->defaultPrinter = NULL;

		GetDefaultPrinter(NULL, &size);
		if (size)
		{
			win_driver->defaultPrinter = (LPWSTR)calloc(size, sizeof(WCHAR));

			if (!win_driver->defaultPrinter)
			{
				win_driver->driver.ReleaseRef(&win_driver->driver);
				return NULL;
			}

			if (!GetDefaultPrinter(win_driver->defaultPrinter, &size))
				win_driver->defaultPrinter[0] = "\0";
		}

(And I did the same kind of error message as for if the win_driver allocation fails ;-) )

Reply:
Or directly `free(win_driver)` instead of `ReleaseRef`?

Reply:
What ever works, probably that the direct free is better as this pointer can't have any other references

Reply:
Done